src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
Corrected my above message, sorry for the mistype: `X` must be a supertype of `R`, `R` can be a composition of `X`